extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/DerivativeDataSourceMetadata.java

```diff
@@ -85,6 +85,12 @@ public boolean matches(DataSourceMetadata other)
     return equals(other);
   }
 
+  @Override
+  public int compareTo(DataSourceMetadata o)
+  {
+    return -1;
```

Comment:
always returning `-1` wouldn't satisfy the Comparable contract here. it would mean A < B and B < A for any two objects A and B implementing this interface. We might need to explain better what the contract implies or throw an exception.